src/cudadev/DataFormats/SoAView.h

```diff
@@ -0,0 +1,375 @@
+    ):                                                                                                                              \
+      _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_CONST_ELEM_MEMBER_INIT, index, VALUE_LIST)                                                \
+       {}                                                                                                                           \
+    _ITERATE_ON_ALL(_DECLARE_VIEW_CONST_ELEMENT_ACCESSOR, ~, VALUE_LIST)                                                            \
+                                                                                                                                    \
+  private:                                                                                                                          \
+    _ITERATE_ON_ALL(_DECLARE_VIEW_CONST_ELEMENT_VALUE_MEMBER, ~, VALUE_LIST)                                                        \
+  };                                                                                                                                \
+                                                                                                                                    \
+  struct element {                                                                                                                  \
+    SOA_HOST_DEVICE_INLINE                                                                                                          \
+    element(size_t index,                                                                                                           \
+      /* Declare parameters */                                                                                                      \
+      _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_ELEMENT_VALUE_ARG, BOOST_PP_EMPTY(), VALUE_LIST)                                          \
+    ):                                                                                                                              \
+      _ITERATE_ON_ALL_COMMA(_DECLARE_VIEW_ELEM_MEMBER_INIT, index, VALUE_LIST)                                                      \
+       {}                                                                                                                           \
+    SOA_HOST_DEVICE_INLINE                                                                                                          \
+    element& operator=(const element& other) {                                                                                      \
+      _ITERATE_ON_ALL(_DECLARE_VIEW_ELEMENT_VALUE_COPY, ~, VALUE_LIST)                                                              \
+      return *this;                                                                                                                 \
+    }                                                                                                                               \
```

Comment:
Got curious, why is the explicit assignment operator needed?

Reply:
This will allow constructs in the form `soa[i] = soa[j]`, copying all members in one go. This expands to:
```C++
    __attribute__(( host )) __attribute__(( device )) __inline__ __attribute__((always_inline)) element& operator=(const element& other) {
      x ( ) = other . x ( ) ;
      y ( ) = other . y ( ) ;
      z ( ) = other . z ( ) ;
[...]
      anotherCount ( ) = other . anotherCount ( ) ;
      description ( ) = other . description ( ) ;
      someNumber ( ) = other . someNumber ( ) ;
      return *this;
    }
```
This is not used so far. And thinking of it, this is going to fail with scalars (now fixed in 7b24d99, adding a constexpr condition).

Reply:
What would the compiler-generated assignment operator do then? (I'm assuming it would be different)

Reply:
In the initial scenario (shown above) we have hand made copy using `T& SoAValue<T>::operator&()` and `T SoAValue<T>::operator() const`. Letting the compiler call `SoAValue<T>::operator=<T2>(const T2&)` member by member would also have achieved the same result. 

This is not true anymore as we now that have the `if constexpr` clause.